Terminology.md

```diff
@@ -0,0 +1,23 @@
+# Terminology for CI systems
+
+## Nouns
+
+* **Job**: Instance of a *job specification*.
+
+* **Job specification**: Description of something to be done by a *worker*.
+
+* **Source**: Somewhere the CI system can fetch the latest version of the source code.
+
+## Verbs
+
+* **Trigger job**: Start an instance of a job, 
+
+## Components
```

Comment:
Although I've imagined these as independent services, some of them might be easier to bundle than to write as complete standalone services.